src/psyclone/psyir/transformations/reference2arrayrange_trans.py

```diff
@@ -156,6 +156,11 @@ def validate(self, node, options=None):
             raise TransformationError(
                 "References to arrays within LBOUND, UBOUND or SIZE "
                 "operators should not be transformed.")
+        if (isinstance(node.parent, IntrinsicCall) and
+                node.parent.routine.name in ["DEALLOCATE"]):
+            raise TransformationError(
+                f"References to arrays within {node.parent.routine.name} "
```

Comment:
I think for debugging purposes it would be useful to include the deallocate statement itself (`fwriter(node.parent)`) in the output text. I guess we'll have to use LazyString as it could happen a lot in an automated script?

Reply:
Done